x-pack/platform/plugins/shared/fields_metadata/server/services/fields_metadata/fields_metadata_client.ts

```diff
@@ -42,17 +45,25 @@ export class FieldsMetadataClient implements IFieldsMetadataClient {
   ): Promise<FieldMetadata | undefined> {
     this.logger.debug(`Retrieving field metadata for: ${fieldName}`);
 
-    // 1. Try resolving from metadata-fields static metadata
-    let field = this.metadataFieldsRepository.getByName(fieldName);
+    // 1. Try resolving from metadata-fields static metadata (highest priority)
+    let field = this.metadataFieldsRepository.getByName(fieldName as AnyFieldName);
 
-    // 2. Try resolving from ecs static metadata
+    // 2. Try searching for the field in the Elastic Package Registry (integration-specific)
+    if (!field && this.hasFleetPermissions(this.capabilities)) {
+      field = await this.integrationFieldsRepository.getByName(fieldName as IntegrationFieldName, {
+        integration,
+        dataset,
+      });
+    }
+
+    // 3. Try resolving from ECS static metadata (authoritative schema)
     if (!field) {
-      field = this.ecsFieldsRepository.getByName(fieldName);
+      field = this.ecsFieldsRepository.getByName(fieldName as AnyFieldName);
     }
```

Comment:
This lookup is most likely to happen than an integration field lookup, and it would have `O(k)` complexity, while looking into the integration repo first might be blocking as it needs to query the EPR, I'd recommend switching back to having the ECS static definition first.